### DISCUSSION TITLE
make v1.23.0 |error: failed to run custom build command for `bellperson v0.24.1`

### DISCUSSION THREAD

**Discussion by peterz3g** [filecoin-project/lotus#10778] [Build and Sync] [answered]
Compiling multihash-derive v0.8.1
error: failed to run custom build command for `bellperson v0.24.1`

Caused by:
  process didn't exit successfully: `/home/jihe/fil_main20230305_bak0428/lotus/extern/filecoin-ffi/rust/target/release/build/bellperson-6598815fc6014bb4/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-cfg=nightly

  --- stderr
  nvcc fatal   : Unknown option '-threads'
  thread 'main' panicked at 'nvcc failed. See the kernel source at /home/jihe/fil_main20230305_bak0428/lotus/extern/filecoin-ffi/rust/target/release/build/bellperson-347b226f04617894/out/22e62c5450c4a1a81043cc36710bfcd865d62713e714b37c0fbffc3d8fbd9e7f.cu', /home/jihe/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/ec-gpu-gen-0.5.2/src/source.rs:660:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `neptune v8.0.0`

Caused by:
  process didn't exit successfully: `/home/jihe/fil_main20230305_bak0428/lotus/extern/filecoin-ffi/rust/target/release/build/neptune-4dfea1994a2d97bb/build-script-build` (exit status: 101)
  --- stderr
  nvcc fatal   : Unknown option '-threads'
  thread 'main' panicked at 'nvcc failed. See the kernel source at /home/jihe/fil_main20230305_bak0428/lotus/extern/filecoin-ffi/rust/target/release/build/neptune-3baffa73fbb9b9ac/out/088ea023b395e88a6e7e207f528719fc6dab1e2dc0851e8afb5c15d90955cc75.cu', /home/jihe/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/ec-gpu-gen-0.5.2/src/source.rs:660:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+ rm -f /ssd1t/filmain/TMPDIR/tmp.ed5amb6fLY
make[1]: *** [Makefile:14：.install-filcrypto] 错误 101
make[1]: 离开目录“/home/jihe/fil_main20230305_bak0428/lotus/extern/filecoin-ffi”
make: *** [Makefile:41：build/.filecoin-install] 错误 2

**Comment by peterz3g** [comment]
Fri Apr 28 08:31:35 2023       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.182.03   Driver Version: 470.182.03   CUDA Version: 11.4     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:21:00.0  On |                  N/A |
|  0%   40C    P8    18W / 270W |    175MiB /  7979MiB |     10%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      1873      G   /usr/lib/xorg/Xorg                 74MiB |
|    0   N/A  N/A      2178      G   /usr/bin/gnome-shell               33MiB |
|    0   N/A  N/A    243471      G   ...859991566675372648,131072       64MiB |
+-----------------------------------------------------------------------------+

**Comment by lionsoul2014** [comment, marked as answer]
看着是 nvcc 版本问题，升级到最新版本的 cuda 试试。
